hadoop-plugin/src/main/groovy/com/linkedin/gradle/dependency/DependencyPattern.groovy

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, VersionPattern 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.linkedin.gradle.dependency;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+
+
+public enum SEVERITY {
+  INFO,
+  WARN,
+  ERROR
+}
+
+
+/**
+ * <p>Java class for DependencyPattern type.
+ *
+ * <p>The following schema fragment specifies the expected content contained within this class.
+ *
+ * <pre>
+ * &lt;type namePattern="DependencyPattern">
+ *   &lt;content>
+ *           &lt;type>
+ *             &lt;content>
+ *                 &lt;sequence>
+ *                   &lt;element name ="namePattern" type="{http://www.w3.org/2001/XMLSchema}string"/>
+ *                   &lt;element name ="groupPattern" type="{http://www.w3.org/2001/XMLSchema}string"/>
+ *                   &lt;element name ="versionPattern" type="{http://www.w3.org/2001/XMLSchema}string"/>
+ *                   &lt;element name ="severity" type="com.linkedin.gradle.dependencyPattern.SEVERITY"/>
+ *                   &lt;element name ="message" type="{http://www.w3.org/2001/XMLSchema}string"/>
+ *                 &lt;/sequence>
+ *             &lt;/content>
+ *           &lt;/type>
+ *   &lt;/content>
+ * &lt;/type>
+ * </pre>
+ *
+ *
+ */
+@XmlAccessorType(XmlAccessType.FIELD)
```

Comment:
You won't need this verbose class if you'd allow specifying the dependency pattern as a groovy map literal, 

http://stackoverflow.com/questions/2212606/groovy-isnt-there-a-stringtomap-out-of-the-box